Assets/MRTK/Services/InputSystem/ConeCastGazeProvider.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Microsoft.MixedReality.Toolkit.Physics;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System.Collections;
+using System.Collections.Generic;
+using Unity.Profiling;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+	public class ConeCastGazeProvider : GazeProvider
```

Comment:
Instead of overriding the existing GazeProvider just to (as far as I can tell) change the gaze pointer type, can we instead make the pointer type configurable on the existing GazeProvider? I might be missing something, but from a quick diff I don't see any other differences. Would help us not have to make a bunch of things `virtual` and `internal`/`public` and leave them as implementation details.

Reply:
Because of how our profile system is set up, changing the pointer type on the GazeProvider itself won't persist those changes across different scenes, as the main profile's only allows the user to configure the GazeProvider type, from which it creates a GazeProvider with the appropriate values. We opted to add a new GazeProvider type, rather than add more settings to the profile and potentially causing users to have to deal with new their profiles suddenly having a new field. This was exacerbated by the fact that the GazeProvider's default pointer type is `InternalGazePointer`, which takes some dependencies on the GazeProvider itself, making it hard to separate out into it's own standalone pointer in the appropriate directory.